enterprise/middleware.py

```diff
@@ -0,0 +1,28 @@
+from enterprise.models import EnterpriseCustomer, UserDataSharingConsentAudit
+from enterprise.tpa_pipeline import (
+    active_provider_enforces_data_sharing,
+    get_consent_status_for_request,
+    get_enterprise_customer_for_request
+)
+class ResetSessionIfPipelineBrokenMiddleware(object):
+    """
+    Middleware signs the user out if they need to provide data sharing consent,
```

Comment:
@haikuginger How did you end up handling this? (Discussion about the middleware you had in this PR and ensuring the pipeline gets completed)